src/Samples/FSSample.cpp

```diff
@@ -0,0 +1,113 @@
+#include "../../headers/Samples/FSSample.h"
+
+FSSample::FSSample(const long SIZE): SIZE(SIZE) {
+    this->data = new double[SIZE];
+}
+
+template<class T>
+void FSSample::baseAdd(const T &value) {
+    this->data[this->index] = ((double) value);
+    ++this->index;
+}
+
+template<class T>
+void FSSample::baseAddCategorical(const T &value, const std::vector<T> &possibleValues) {
+    for(auto it = possibleValues.begin(); it != possibleValues.end(); ++it)
+    {
+        if(*it == value)
+            this->data[this->index] =(double) 1;
+        else
+            this->data[this->index] =(double) 0;
+        ++this->index;
```

Comment:
Increment at the end by the size of the possible values list instead of incrementing in the loop